compose.yaml

```diff
@@ -0,0 +1,52 @@
+services:
+  db:
+    build:
+      context: .
+      dockerfile: dockerfiles/Dockerfile.db
+    image: dtbase_db
+    restart: always
+    ports:
+      - "5432:5432"
+    env_file:
+      - .env
+    environment:
+      - POSTGRES_USER=${DT_SQL_USER}
+      - POSTGRES_PASSWORD=${DT_SQL_PASS}
+      - POSTGRES_DB=${DT_SQL_DBNAME}
+    volumes:
+      - app_data:/var/lib/postgresql/data
+  backend:
+    build:
+      context: .
+      dockerfile: dockerfiles/Dockerfile.backend
+    image: backend
+    ports:
+      - "5000:5000"
+    depends_on:
+      - db
+    env_file:
+      - .env
+    environment:
```

Comment:
How do `env_file` and `environment` interact? Is one for build-time and the other for run-time? Or does one pick values from the other or something like that?

Reply:
The `env_file` is the source of the variables. `environment` are the variables being used by the container. We only need environment here because some of our environment variables are named things that make more sense in the dtbase context inthe .env file but need to be called something else with respect to the container. 

I.E: `POSTGRES_USER=${DT_SQL_USER}`